src/AutoRest.CSharp.V3/Pipeline/Extensions.cs

```diff
@@ -58,15 +66,29 @@ internal static class Extensions
             _ => null
         };
 
-        private static readonly Func<string, bool, string?> NumberSerializer =
-            (vn, nu) => $"writer.WriteNumberValue({vn}{(nu ? ".Value" : string.Empty)});";
-        private static Func<string, bool, string?> StringSerializer(bool includeToString = false) =>
-            (vn, nu) => $"writer.WriteStringValue({vn}{(includeToString ? ".ToString()" : string.Empty)});";
+        private static readonly Func<string, bool, FrameworkTypeReferenceFormat, string?> NumberSerializer =
+            (vn, nu, f) => $"writer.WriteNumberValue({vn}{(nu ? ".Value" : string.Empty)});";
+        private static Func<string, bool, FrameworkTypeReferenceFormat, string?> StringSerializer(bool includeToString = false) =>
+            (vn, nu, f) => $"writer.WriteStringValue({vn}{(includeToString ? ".ToString()" : string.Empty)});";
+
+        private static string? ToFormatSpecifier(this FrameworkTypeReferenceFormat format) => format switch
```

Comment:
Let's unify this with the WriteSerializationFormat

Reply:
So, the actual formats in WriteSerializationFormat are different than this. Do header/query params have a different format than the body properties?

Reply:
They are not different, they are all endup calling into https://github.com/Azure/autorest.csharp/blob/feature/v3/src/assets/Azure.Core.Shared/TypeFormatters.cs#L17 with formats exactly like here.

Reply:
I was confused that there are specifiers in WriteSerializationFormat that are specifiers for the extensions, not specifiers for the ToString on DateTimeOffset. My bad.

Reply:
So, this is a little complicated right now. Basically, I put the format specifier code (switch statement) from WriteSerializationFormat here. Why? Because, right now, the code to do this serialization logic is in Extensions. And, Extensions doesn't use the actual CodeWriter; it just returns strings. When we move all the Json-specific serialization logic out of Extensions, we can use the same WriteSerializationFormat method. For now, I did the best I can without that refactoring.